extension/chrome/settings/modules/contacts.ts

```diff
@@ -23,6 +26,54 @@ Catch.try(async () => {
 
   BrowserMsg.listen(tabId); // set_css
 
+  const attUI = new AttUI(() => Promise.resolve({ size_mb: 5, size: 5 * 1024 * 1024, count: 1 }));
+  attUI.initAttDialog('fineuploader', 'fineuploader_button');
+  attUI.setAttAddedCb(async (file: Att) => {
+    attUI.clearAtts();
+    await handleKeyfileSelected(file);
+  });
+
+  const handleKeyfileSelected = async (file: Att) => {
+    try {
+      const rawFile = file.getData();
+      const stringFile = new TextDecoder("utf-8").decode(rawFile);
+      const { blocks } = await Pgp.armor.detectBlocks(stringFile);
+
+      const keys: OpenPGP.key.Key[] = [];
+      if (blocks.length > 0 && blocks.some(block => block.type === "publicKey")) {
+        for (const block of blocks) {
+          if (block.type === "publicKey") {
+            const result = await openpgp.key.readArmored(block.content);
+            if (result.err) {
```

Comment:
Can collect errs into array like you do with keys, to prevent code duplication, then check errs at the end with a single err handling block.